### ISSUE TITLE
[HLSL] Generate AST for Root Signatures

### ISSUE THREAD

**Issue by inbelic** [llvm/llvm-project#119011]
This issue handles "plugging-in" the root signature parser into the generation of the AST node. It consists of initializing an `HLSLRootSignatureAttr` with the parsed `HLSLRootElements`.

Root signatures are defined as a list of "root elements" and so the in-memory representation will retain this form.

This is dependent on #120472.

AC:
- [ ] Define `HLSLRootSignatureAttr` in `Attr.td`
- [ ] Use `AdditionalMembers` to hold an ArrayRef of `HLSLRootElement`s
- [ ] Define `handleHLSLRootSignature` in `SemaHLSL.h`
- [ ] Implement `handleHLSLRootSignature` in `SemaHLSL.cpp` to simply invoke the parser and construct the Root Elements on the ASTContext, then set the `AdditionalMembers` to refence that
- [ ] Add test of a practical root signature example generated AST with `ast-dump` in `AST/HLSL/RootSignatures-AST.hlsl`

**Comment by inbelic** [comment]
Self-note for information regarding default values: https://learn.microsoft.com/en-us/windows/win32/direct3d12/specifying-root-signatures-in-hlsl

**Comment by damyanp** [comment]
Feedback from refinement: split this into two issues - one for parsing the root signature string, and then another for adding the attribute and hooking it into the parser.

**Comment by damyanp** [comment]
Needs some information about how ast-dump will be modified to allow for testing.

**Comment by inbelic** [comment]
Exploring the ability to place the parsing into `ParseCXXDecl.cpp`

**Comment by inbelic** [comment]
This will be blocked on https://github.com/llvm/llvm-project/issues/138025 being partial complete such that we can test the correct decl elements are being copied over.
